src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentActor.scala

```diff
@@ -146,8 +145,8 @@ private class DeploymentActor(
   private def killInstancesIfNeeded(instancesToKill: Seq[Instance]): Future[Done] = async {
     logger.debug("Kill instances {}", instancesToKill)
     val changeGoalsFuture = instancesToKill.map(i => {
-      if (i.hasReservation) instanceTracker.setGoal(i.instanceId, Goal.Stopped)
-      else instanceTracker.setGoal(i.instanceId, Goal.Decommissioned)
+      if (i.hasReservation) scheduler.stop(i.instanceId)
+      else scheduler.decommission(i.instanceId)
     })
     await(Future.sequence(changeGoalsFuture))
     await(killService.killInstances(instancesToKill, KillReason.DeploymentScaling))
```

Comment:
I could move this call to `Scheduler.stop` and `Scheduler.decommission`. What do you think?

Reply:
yes, I think we should do that...

Reply:
second that